var/spack/repos/builtin/packages/boost/package.py

```diff
@@ -462,24 +462,24 @@ def flag_handler(self, name, flags):
 
     def determine_toolset(self, spec):
         toolsets = {
-            "g++": "gcc",
-            "icpc": "intel",
-            "icpx": "intel",
-            "clang++": "clang",
-            "armclang++": "clang",
-            "xlc++": "xlcpp",
-            "xlc++_r": "xlcpp",
-            "pgc++": "pgi",
-            "nvc++": "pgi",
-            "FCC": "clang",
+            "%gcc": "gcc",
+            "%intel": "intel",
+            "%oneapi": "intel",
+            "%clang": "clang",
+            "%arm": "clang",
+            "%xl": "xlcpp",
+            "%xl_r": "xlcpp",
+            "%pgi": "pgi",
+            "%nvhpc": "pgi",
+            "%fj": "clang",
         }
 
         if spec.satisfies("@1.47:"):
-            toolsets["icpc"] += "-linux"
-            toolsets["icpx"] += "-linux"
+            toolsets["%intel"] += "-linux"
+            toolsets["%oneapi"] += "-linux"
 
         for cc, toolset in toolsets.items():
-            if cc in self.compiler.cxx_names:
+            if cc in self.spec.satiisfies(cc):
```

Comment:
1. Typo in 'satisfies'
2. Should this just be `if self.spec.satisfies(cc)`?

Reply:
Thanks for catching this typo. I was planning to extract package changes from this PR and ping people there.